### ISSUE TITLE
1.0.0-rc2 does not work with vtk 6.0

### ISSUE THREAD

**Issue by andreasbuhr** [libMesh/libmesh#1009]
Hi there,

I am testing the 1.0.0-rc2 and have trouble installing on my Linux Mint 17.2 (based on Ubuntu trusty aka 14.04). Linux Mint 17.2 ships with vtk6.0 and the configure script does not accept it, because "vtkIOParallelXML-6.0" is not found. There does not exist anything named "vtkIOParallelXML*" in /usr/lib. On a fresh Ubuntu 16.04 installation, with vtk6.2, everything works fine.
Here is the output of "configure" on my Linux Mint 17.2: 
[configure.out.txt](https://github.com/libMesh/libmesh/files/331754/configure.out.txt)
Here is the config.log from the same run:
[config.log.txt](https://github.com/libMesh/libmesh/files/331755/config.log.txt)

Just for comparison, here is the output of "configure" on Ubuntu 16.04: 
[configure.out.txt](https://github.com/libMesh/libmesh/files/331764/configure.out.txt)
and the correspondig config.log:
[config.log.txt](https://github.com/libMesh/libmesh/files/331765/config.log.txt)

Is libmesh supposed to work with vtk6.0? In that case it would be good to have a test which does not require vtkIOParallelXML. Or isn't it supposed to work with vtk6.0? In that case, an error message like "at least vtk 6.2 is required" would be nice.

**Comment by jwpeterson** [comment]
> because "vtkIOParallelXML-6.0" is not found

None of the libraries is found, not just vtkIOParallelXML.so:
```
configure:39350: mpicxx -o conftest  -std=gnu++11  -I/usr/include/vtk-6.0  conftest.cpp  -Wl,-rpath,/usr/lib -L/usr/lib -lvtkIOCore -lvtkCommonCore -lvtkCommonDataModel                                    -lvtkFiltersCore -lvtkIOXML -lvtkImagingCore                                    -lvtkIOImage -lvtkImagingMath -lvtkIOParallelXML >&5
/usr/bin/ld: cannot find -lvtkIOCore
/usr/bin/ld: cannot find -lvtkCommonCore
/usr/bin/ld: cannot find -lvtkCommonDataModel
/usr/bin/ld: cannot find -lvtkFiltersCore
/usr/bin/ld: cannot find -lvtkIOXML
/usr/bin/ld: cannot find -lvtkImagingCore
/usr/bin/ld: cannot find -lvtkIOImage
/usr/bin/ld: cannot find -lvtkImagingMath
/usr/bin/ld: cannot find -lvtkIOParallelXML
collect2: error: ld returned 1 exit status
```

A similar [issue](https://sourceforge.net/p/libmesh/mailman/message/34522399) was reported back in October 2015 for Fedora and Arch Linux.  It looks like there is a gap in our vtk.m4 file specifically for version 6.0 of VTK (I guess no one ever tried to use this version...), we have special logic for 5.x, 6.1, and 6.2+, but nothing for 6.0, so the 6.0 case is accidentally being caught up in the 6.2+ case...

A quick thing you could try would be to change the last part of the following line in vtk.m4:
```
         elif (test $vtkmajor -eq 6 -a $vtkminor -eq 1); then
```

to "$vtkminor -le 1", and rerun bootstrap.  Then try configuring again and see if it works...

**Comment by andreasbuhr** [comment]
Hi, thanks a lot for your help. However, I am still a little confused, because the error you cite also occurs on my Ubuntu 16.04 machine. Because it worked on Ubuntu 16.04 and not on the Mint 17.2, I compared the config.log files. The error which occurs on Mint but not on Ubuntu is the one with "vtkIOParallelXML".

**Comment by jwpeterson** [comment]
Right, the other error occurs because we are checking for an alternate naming scheme for the shared libs (both with and without the version numbers appended) and is normal.

**Comment by andreasbuhr** [comment]
Thanks a lot for this clarification. The code change you proposed works fine on my machine. I created a pull request, in case you want to merge it to master: [pull request](https://github.com/libMesh/libmesh/pull/1010)

Thanks again for your quick help! :-)

**Comment by jwpeterson** [comment]
Fixed in #1010
